app/src/main/java/mozilla/lockbox/store/AutoLockStore.kt

```diff
@@ -33,64 +38,80 @@ class AutoLockStore(
 
     private val compositeDisposable = CompositeDisposable()
     private lateinit var preferences: SharedPreferences
-    private lateinit var currentBootID: String
 
-    val lockRequired: Observable<Boolean> = ReplaySubject.createWithSize(1)
+    private val currentBootID: String
+        get() = lockingSupport.currentBootId
+
+    var lockingSupport: LockingSupport = SystemLockingSupport()
+
+    val lockRequired: Observable<Boolean> = PublishSubject.create()
 
     init {
-        lifecycleStore.lifecycleFilter
+        lifecycleStore.lifecycleEvents
             .filter { it == LifecycleAction.Foreground }
             .map { lockCurrentlyRequired() }
             .subscribe(lockRequired as Subject)
 
-        lifecycleStore.lifecycleFilter
-            .filter { it == LifecycleAction.Background }
+        Observables.combineLatest(lifecycleStore.lifecycleEvents, dataStore.state)
+            .filter { it.first == LifecycleAction.Background && it.second != DataStore.State.Locked }
             .switchMap { settingStore.autoLockTime.take(1) }
             .subscribe(this::updateNextLockTime)
             .addTo(compositeDisposable)
+
+        dispatcher.register
+            .filter { it == DataStoreAction.Lock }
+            .subscribe {
+                this.backdateNextLockTime()
+            }
+            .addTo(compositeDisposable)
+
+        // wait to store the currentBootId until we can be sure that we've checked it once
+        lockRequired
+            .take(1)
+            .subscribe { storeCurrentBootId() }
+            .addTo(compositeDisposable)
     }
 
     override fun injectContext(context: Context) {
         preferences = PreferenceManager.getDefaultSharedPreferences(context)
+    }
 
-        currentBootID = FileReader().readContents(Constant.App.bootIDPath)
-
-        preferences
-            .edit()
-            .putString(Constant.Key.bootID, currentBootID)
-            .apply()
+    private fun backdateNextLockTime() {
+        storeAutoLockTimerDate(lockingSupport.systemTimeElapsed - 1)
```

Comment:
What are the benefits to backdating?  Naively it seems like setting it to`0` would have been sufficient.